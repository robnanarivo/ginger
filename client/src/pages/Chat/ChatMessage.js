import React from 'react';
import { Grid, Typography } from '@mui/material';

const ChatMessage = ({
  chatter,
  content,
  timestamp,
  isSelf,
}) => {
  let align = 'left';
  let sender = chatter;
  if (isSelf) {
    align = 'right';
    sender = 'You';
  }
  return (
    <Grid container sx={{ p: 2 }}>
      <Grid item xs={12} align={align}>
        <Typography sx={{ fontSize: 14, color: 'gray' }}>{sender}</Typography>
        <Typography sx={{ fontSize: 18 }}>{content}</Typography>
        <Typography sx={{ fontSize: 14, color: 'gray' }}>{timestamp}</Typography>
      </Grid>
    </Grid>
  );
};

export default ChatMessage;
