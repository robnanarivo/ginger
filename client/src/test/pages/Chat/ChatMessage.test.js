import React from 'react';
import renderer from 'react-test-renderer';
import ChatMessage from '../../../pages/Chat/ChatMessage';
import 'jest-styled-components';

describe('ChatMessage unit test', () => {
  it('ChatMessage Snapshot test', () => {
    const tree = renderer
      .create(<ChatMessage sender="You" isSelf timestamp="20211202" content="Wow" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
