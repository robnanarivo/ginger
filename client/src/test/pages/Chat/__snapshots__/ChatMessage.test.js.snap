// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage unit test ChatMessage Snapshot test 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-1fkyx8n-MuiGrid-root"
>
  <div
    align="right"
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-vvr8k2-MuiTypography-root"
    >
      You
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-qscxtw-MuiTypography-root"
    >
      Wow
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-vvr8k2-MuiTypography-root"
    >
      20211202
    </p>
  </div>
</div>
`;
